api/turing/validation/validator.go

```diff
@@ -46,6 +49,24 @@ func NewValidator(expSvc service.ExperimentsService) (*validator.Validate, error
 	return instance, nil
 }
 
+func validateEnsemblerStandardConfig(sl validator.StructLevel) {
```

Comment:
Had to go for a `StructLevel` validation in the end, which churns out error logs with the `CreateOrUpdateRouterRequest.Config.Ensembler.StandardConfig.[...]` path instead of the `FieldLevel` validation option as discussed earlier. Apparently the go-playground validator is [unable](https://github.com/go-playground/validator/issues/470) to perform `FieldLevel` validations on structs, even with a custom validation function defined.

This isn't too much of an issue, except for the enforced error logging path, which indirectly forces us to duplicate the error messages returned e.g. instead of returning "EnsemblerStandardConfig cannot have both ExperimentMappings and RouteNamePath set" with the `CreateOrUpdateRouterRequest.Config.Ensembler.StandardConfig` path, we can only return "ExperimentMappings cannot be set when RouteNamePath is set" with the `CreateOrUpdateRouterRequest.Config.Ensembler.StandardConfig.ExperimentMappings` path AND "RouteNamePath cannot be set when ExperimentMappings is set" with the `CreateOrUpdateRouterRequest.Config.Ensembler.StandardConfig.RouteNamePath` path.

Reply:
Makes sense. As for the duplicate error messages, I think we can just log the message once instead of calling `sl.ReportError` twice under each check. I guess it's ok to associate the error to either field `ExperimentMappings` or `RouteNamePath` ? I'm also ok with leaving it as it is - no strong opinion.

Reply:
That makes sense; I've removed the duplicated messages and decided to keep the message concerning `ExperimentMappings` in both instances (where both fields are set or neither fields are set), and drop the one concerning `RouteNamePath`.